compliance/tags/azure_rg_tags/azure_resource_group_tags.pt

```diff
@@ -0,0 +1,303 @@
+escalation "tag_resources" do
+  run "tag_resources", data, $param_azure_tenant_id, $param_azure_sub, $param_tag_key
+end
+
+define tag_resources($data, $param_azure_tenant_id, $param_azure_sub, $param_tag_key) return $all_responses do
+
+  call get_access_token($param_azure_tenant_id) retrieve $access_token
+
+  $providers_response = http_request(
+    verb: "get",
+    host: "management.azure.com",
+    https: true,
+    href: "/subscriptions/"+$param_azure_sub+"/providers",
+    query_strings: {
+      "api-version": "2018-05-01"
+    },
+    headers: {
+      "cache-control": "no-cache",
+      "content-type": "application/json",
+      "authorization": "Bearer " + $access_token
+    }
+  )
+  $providers_array = $providers_response["body"]["value"]
+  $$log = []
+  $all_responses = []
+  foreach $item in $data do
+    $$log << "ID: "+$item["id"]
+    $namespace = $item["namespace"]
+    $$log << "Namespace: "+$namespace
+    if $namespace =~ "Classic"
+      # Skip
+      $$log << "Skipping due to Azure Classic"
+    else
+      $resource_type = $item["resource_type"]
+      $$log << "Resource Type: "+$resource_type
+      $namespace_obj = select($providers_array, {"namespace": $namespace})
+      if $namespace_obj == []
+        $namespace_obj = select($providers_array, {"namespace": downcase($namespace)})
+      end 
+      $resource_type_obj = select($namespace_obj[0]["resourceTypes"], {"resourceType": $resource_type})
+      $api_versions = $resource_type_obj[0]["apiVersions"]
+      $api_version = first($api_versions)
+      $$log << "API Version: "+$api_version
+
+      $resource_response = http_request(
+        verb: "get",
+        host: "management.azure.com",
+        https: true,
+        href: $item["id"],
+        query_strings: {
+          "api-version": $api_version
+        },
+        headers: {
+          "cache-control": "no-cache",
+          "content-type": "application/json",
+          "authorization": "Bearer " + $access_token
+        }
+      )
+
+      $all_responses << $resource_response
+
+      $new_object = $resource_response["body"]
+      $new_tags = {}
+      if $item["tag"] == "none"
+        $new_object["tags"] = {}
+      else
+        $new_tags = $resource_response["body"]["tags"]
+      end 
+      $new_object["tags"][$param_tag_key] = $item["resource_group"]
+      $new_tags[$param_tag_key] = $item["resource_group"]
+      sub on_error: handle_error($update_resource_response) do 
+        $update_resource_response = http_request(
+          #verb: "put",
+          verb: "patch",
+          host: "management.azure.com",
+          https: true,
+          href: $item["id"],
+          query_strings: {
+            "api-version": $api_version
+          },
+          headers: {
+            "cache-control": "no-cache",
+            "content-type": "application/json",
+            "authorization": "Bearer " + $access_token
+          },
+          #body: $new_object
+          body: {"tags": $new_tags}
```

Comment:
I am guessing you have to reset tags here, can't just add an additional tag. My 2 Cents you are already fetching the resource tags in the javascript data source. Add an additional field to the results to store existing tags and reuse it here while adding a new tag instead of an additional API call to fetch tags.

Reply:
good call! fixed.